### PR TITLE
feat: add figure for genie patients matching at least 1 variant to a kb

### PR DESCRIPTION
**Pull request by jarbesfeld**
Display output from GENIE analysis in Results Tab

### CODE REVIEW

**Comment by jarbesfeld** [comment]
@korikuzma Would I pull from main in my terminal or in the pull request? I don't see an option to pull from main here.

**Comment by korikuzma** [comment]
> @korikuzma Would I pull from main in my terminal or in the pull request? I don't see an option to pull from main here.

You need to resolve the conflicts

**Comment by jarbesfeld** [comment]
@korikuzma Can I resolve the conflicts using the web editor? This is what I see in the web editor:

<<<<<<< patient-graph
## Results

Example Results- Patient Matching with GENIE

Variant normalization allows patient data from AACR Project GENIE to be matched to normalized variants in the CIViC, MOA, and ClinVar knowledgebases.
=======
### Running Notebooks in Visual Studio Code (VS Code)

[VS Code](https://code.visualstudio.com/) is a lightweight source code editor for Windows, Linux, and macOS.

1. Download VS Code [here](https://code.visualstudio.com/Download)
2. Open a notebook and click `Select Kernel` at the top right. Select the option where the path is `venv/3.11/bin/python`. See [here](https://code.visualstudio.com/docs/datascience/jupyter-kernel-management) for more information on managing Jupyter Kernels in VS Code.
3. Run the notebooks

## Analysis with macOS Environments

These notebooks were run using these macOS specs:

| Model Year | CPU Architecture | Total RAM | Hard drive capacity |
| --- | --- | --- | --- |
| 2019 | 2.6 GHz 6-Core Intel Core i7 | 32 GB | 1 TB |
| 2021 | M1 Pro | 32 GB | 1 TB |
>>>>>>> main

I think this is my first time resolving a merge conflict so I wasn't sure where I would move the results section to?

**Comment by korikuzma** [comment]
@jarbesfeld ask @wesleygoar

Docs on resolving conflict in GH: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-on-github

**Comment by wesleygoar** [comment]
Am I being asked a question or just doing a review? @jarbesfeld

**Comment by jarbesfeld** [comment]
@wesleygoar I think review. I think I fixed the merge conflict

**Comment by korikuzma** [comment]
@jarbesfeld was asking "I wasn't sure where I would move the results section to?" @wesleygoar

File: README.md [hunk 1]
```diff
@@ -226,6 +226,10 @@ This section provides information about the notebooks and the order that they sh
     * [analysis/performance_analysis/merged_performance_analysis.ipynb](./analysis/performance_analysis/merged_performance_analysis.ipynb)
       * Analysis on Variation Normalizer performance on CIViC, MOA, and ClinVar
 
+## Results
```
Comment (korikuzma) [inline README.md:229]: `Running Notebooks in Visual Studio Code (VS Code)` is not a sub section for Results. 

```suggestion
### Notebook Results
```

File: README.md [hunk 2]
```diff
@@ -226,6 +226,10 @@ This section provides information about the notebooks and the order that they sh
...
+![Example Results- Patient Matching with GENIE](./analysis/genie/variant_analysis/genie_patient_matching.png)
```
Comment (korikuzma) [inline README.md:230]:
```suggestion

![Patient Matching with GENIE](./analysis/genie/variant_analysis/genie_patient_matching.png)
```

File: README.md [hunk 3]
```diff
@@ -226,6 +226,10 @@ This section provides information about the notebooks and the order that they sh
...
+Variant normalization allows patient data from AACR Project GENIE to be matched to normalized variants in the CIViC, MOA, and ClinVar knowledgebases.
+
```
Comment (korikuzma) [inline README.md:232]:
```suggestion

```

**Review by korikuzma — CHANGES_REQUESTED** [review]

**Comment by jarbesfeld** [comment]
@korikuzma I think this is probably a simple question, but after resolving merging conflicts, how would I pull in the latest changes to my local branch? I tried `git pull` on my patient-graph branch, and got, where the README was unchanged:
 Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> patient-graph

**Comment by wesleygoar** [comment]
> @jarbesfeld asked "I wasn't sure where I would move the results section to?" @wesleygoar

@jarbesfeld you can either keep results where it is or move it all the way to the top section and say something like, "After running these notebooks users will be able to create figures such as this that demonstrate the results of this analysis."

**Comment by wesleygoar** [comment]
Screenshot 2023-11-22 at 10 20 56 AM
@jarbesfeld if you move to top put it in this section.

**Comment by korikuzma** [comment]
> @korikuzma I think this is probably a simple question, but after resolving merging conflicts, how would I pull in the latest changes to my local branch? I tried `git pull` on my patient-graph branch, and got, where the README was unchanged: Please specify which branch you want to merge with. See git-pull(1) for details.
> 
> ```
> git pull <remote> <branch>
> ```
> 
> If you wish to set tracking information for this branch you can do so with:
> 
> ```
> git branch --set-upstream-to=origin/<branch> patient-graph
> ```

@jarbesfeld did you try running any of those prompts?

**Comment by korikuzma** [comment]
> Screenshot 2023-11-22 at 10 20 56 AM @jarbesfeld if you move to top put it in this section.

@wesleygoar I think this is a good "hook"

**Comment by jarbesfeld** [comment]
@korikuzma I tried running the first command, but I was unsure what would go in <remote>. I know <branch> would be patient graph.

@wesleygoar I can move results to the top

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 4]
```diff
@@ -1291,48 +1321,163 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Create Grouped Bar Plot for Table 2\n",
+    "summary_df_patient_plot = summary_df_patient.T\n",
+    "summary_df_patient_plot = summary_df_patient_plot[\n",
+    "    [\"MOAlmanac\", \"CIViC\", \"ClinVar\", \"All Sources\"]\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1394]: @wesleygoar how would you feel about this

```suggestion
    "    [\"MOAlmanac\", \"CIViC\", \"ClinVar\", \"All KBs\"]\n",
```

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 5]
```diff
@@ -1239,32 +1265,36 @@
     "    :param n, d: Two integers\n",
     "    :return: A string reporting the percentage\"\"\"\n",
     "    percent = round(100 * n / d, 1)\n",
-    "    return f\"{n} / {d} ({percent}%)\"\n",
+    "    return percent\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1268]: You need to update the return type and docstring. For params, you create two separate lines: 

```python
:param param1:
:param param2:
```

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -4,6 +4,11 @@ This repo contains analysis notebooks used in the _The Clinical Genomic Variatio
 
 Small output files can be found in this repo. Larger files can be found in our public s3 bucket: `s3://nch-igm-wagner-lab-public/variation-normalizer-manuscript/`. There are notebooks that provide functions for programmatically downloading files from the s3 bucket.
 
+### Results
+![Patient Matching with GENIE](./analysis/genie/variant_analysis/genie_patient_matching.png)
+
+Variant normalization allows patient data from AACR Project GENIE to be matched to normalized variants in the CIViC, MOA, and ClinVar knowledgebases.
+
```
Comment (korikuzma) [inline README.md:11]: I would move this to the beginning like @wesleygoar had mentioned as a possible place

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 7]
```diff
@@ -1228,43 +1254,48 @@
...
   {
    "cell_type": "code",
-   "execution_count": 47,
+   "execution_count": 48,
    "id": "efc9e0ad",
    "metadata": {},
    "outputs": [],
    "source": [
     "# Helper function\n",
     "def compute_percent(n: int, d: int) -> str:\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1263]: You need to update the return type
```suggestion
    "def compute_percent(n: int, d: int) -> float:\n",
```

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 8]
```diff
@@ -1228,43 +1254,48 @@
...
     "    \"\"\"Format percentage\n",
-    "    :param n, d: Two integers\n",
+    "    :param n: An integer\n",
+    "    :param d: An integer\n",
     "    :return: A string reporting the percentage\"\"\"\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1267]: You also need to update the return in the docstring

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 9]
```diff
@@ -1228,43 +1254,48 @@
...
     "    percent = round(100 * n / d, 1)\n",
-    "    return f\"{n} / {d} ({percent}%)\"\n",
+    "    return percent\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1269]: you can just return and not store in variable

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 10]
```diff
@@ -1228,43 +1254,48 @@
...
     "\n",
     "\n",
     "# Table 2 - Patient Matching. The percentages are computed using the total number of patients in the GENIE dataset.\n",
-    "summary_df_patient = pd.DataFrame(columns=[\"Protein\", \"Genomic\"])\n",
-    "summary_df_patient.loc[\"(%) of Patients with Normalized Variants Across Sources\"] = [\n",
+    "summary_df_patient = pd.DataFrame(columns=[\"Protein\", \"Genomic\", \"Protein or Genomic\"])\n",
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1273]: This section always uses `total_patients` in the `compute_percent` function. Just use that value in the function if it's not changing. Function parameters are good for values that are changing between calls

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -1,5 +1,10 @@
 # Variation Normalizer Manuscript
 
+### Results
```
Comment (korikuzma) [inline README.md:3]:
```suggestion
```

File: README.md [hunk 12]
```diff
@@ -1,5 +1,10 @@
...
+![Patient Matching with GENIE](./analysis/genie/variant_analysis/genie_patient_matching.png)
+
+Variant normalization allows patient data from AACR Project GENIE to be matched to normalized variants in the CIViC, MOA, and ClinVar knowledgebases.
```
Comment (korikuzma) [inline README.md:6]: Move this after the data in repo / s3 section. And have the text go before the image and state like "as shown in the following figure"

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb (hunk 10 above)
Comment (korikuzma) [reply analysis/genie/variant_analysis/genie_search_analysis.ipynb:1272]: @jarbesfeld I didn't mean hard code the value, I meant remove the parameter in the function and just use the `total_patients` variable there

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: analysis/genie/variant_analysis/genie_search_analysis.ipynb [hunk 13]
```diff
@@ -1291,48 +1320,163 @@
...
+    "for container in patient_plot.containers:\n",
+    "    patient_plot.bar_label(container)\n",
+    "patient_plot.legend(loc=\"upper right\", ncols=4)\n",
+    "patient_plot.set_ylim(0, 110)\n",
+    "patient_plot.set_ylabel(\"Percent of Patients with Match\")\n",
+    "patient_plot.set_xlabel(\"Variant Type\")\n",
+    "patient_plot.set_title(\n",
+    "    \"Percent of GENIE Patients with At Least One Variant Match to a Knowledgebase\"\n",
+    ")\n",
+    "patient_plot.figure.savefig(\"genie_patient_matching.png\", bbox_inches=\"tight\")"
```
Comment (korikuzma) [inline analysis/genie/variant_analysis/genie_search_analysis.ipynb:1407]: For better resolution

```suggestion
    "patient_plot.figure.savefig(\"genie_patient_matching.png\", bbox_inches=\"tight\", dpi=1000)"
```

**Review by korikuzma — CHANGES_REQUESTED** [review]

**Review by korikuzma — APPROVED** [review]
@wesleygoar your turn

**Comment by wesleygoar** [comment]
@jarbesfeld or @korikuzma can you please add this statement above the line 7. The result comes sharply out of the blue without any context. 
"After running these notebooks, users will be able to create figures such as this that demonstrate the results of this analysis."

**Review by wesleygoar — APPROVED** [review]

**Pull request merged**